monolith-discovery.typ

```diff
@@ -0,0 +1,19 @@
+= Monolith Discovery <Chapter::MonolithDiscovery>
+
+To connect, the load balancer must must know the network addresses of the monoliths, and this process of figuring out where the monoliths are is discovery.
+
+The first case is connecting to the instance of the monolith on fly.io. Before connecting the user inputs the port that the monolith should listen on for the load balancer connection and a string representing the fly app. These properties are both encapsulated by the FlyDiscoveryCongig struct.
+
+To discover the monolith, the configuration and a query are passed into the discoverer struct, FlyMonolithDiscoverer. In order to connect to the instance on fly.io, the query string is formatted and then a connection attempts. An IPv6 lookup is created using the query from before and a response is awaited asynchronously. The discovery process uses the results of the lookup and connects the balancer.
```

Comment:
This is better, but this doesn't describe the process in a general manner. The discoverers use various methods to discover monoliths, and those are implementation details. Those methods are worth mentioning, but as supporting material to the more generic `MonolithDiscovery` trait.

Traits are fairly similar to how class/interface inheritance works. You can have type T that implements trait U, much like you can have type T extend the interface U, (also fyi there are no "base classes" in rust).

What I'm more getting at is that you should explain the general idea and goal first, and then you can dive deeper into the implementations of the discoverers.